frontend/src/pages/CollaborationPage.js

```diff
@@ -73,27 +77,49 @@ const CollaborationPage = () => {
 
     handleLeaveRoomConfirmationCloseDialog()
     handleReviewPartnerOpenDialog()
-  }, [location.state.qnsid, location.state.room])
+  }, [location.state])
 
-  const checkExpiry = useCallback(async (room_id) => {
-    try {
-      const res = await findRoomService({ room_id: room_id })
-      if (res.data && JSON.stringify(res.data) !== '{}') {
-        expirationCheck(
-          res.data.datetime,
-          async () => {
-            console.log(`room: ${room_id} expired`)
-            // await deleteRoomService(room_id)
-            // navigate(homeUrl, { replace: true })
-            leaveRoom()
-          },
-          () => {}
-        )
+  const getTimeRemaining = useCallback(() => {
+    const now = moment()
+    const end = moment(room.datetime)
+    const diff = now.diff(end, 'minutes')
+    const timeLeft = diff < 30 ? MAXTIME - diff : 0
```

Comment:
Since `MAXTIME` is 30, perhaps you can change 30 to `MAXTIME`?
i.e. `const timeLeft = diff < MAXTIME ? MAXTIME - diff : 0`

Reply:
And if you are changing from `MAXTIME` to `MAX_TIME_MINUTES`, then it will be
`const timeLeft = diff < MAX_TIME_MINUTES ? MAX_TIME_MINUTES - diff : 0`